client/lib/signup/step-actions/index.js

```diff
@@ -438,7 +438,8 @@ export function createAccount(
 			( response && response.user_id ) ||
 			userData.ID;
 
-		const email = ( response && response.email ) || ( userData && userData.user_email );
+		const email =
+			( response && response.email ) || ( userData && ( userData.email || userData.user_email ) );
```

Comment:
Just a nitpick from a code-style perspective (can't really comment on the rest of what's going on in this PR) but you might consider using the [nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator) and [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) to simplify this expression.

```suggestion
		const email =
			response?.email ?? userData?.email ?? userData?.user_email ?? '';
```

I'm assuming `''` (empty string) is a safer default than `undefined` or `null`. Which is probably most likely what would happen here. That assumption might be wrong though 😊

Reply:
I think we'd rather have `null` or `undefined` so it breaks if it's not set, vs. sending completely wrong data. Someone should probably create an HDV rule to detect this kind of breakage in the future.